Google.PowerShell/Container/GkeClusterCmdlets.cs

```diff
@@ -0,0 +1,163 @@
+
+        protected override void ProcessRecord()
+        {
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.AllZone:
+                    // "-" represents all zone.
+                    WriteObject(GetClustersByZone(Project, "-"), true);
+                    break;
+                case ParameterSetNames.ByZone:
+                    WriteObject(GetClustersByZone(Project, Zone), true);
+                    break;
+                case ParameterSetNames.ByName:
+                    WriteObject(GetClustersByName(Project, Zone, ClusterName), true);
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
+        }
+
+        /// <summary>
+        /// Returns clusters that have the names in clusters array in zone 'zone' in project 'project'.
+        /// </summary>
+        private IEnumerable<Cluster> GetClustersByName(string project, string zone, string[] clusters)
+        {
+            foreach (var cluster in clusters)
+            {
+                Cluster result = null;
+                try
+                {
+                    ProjectsResource.ZonesResource.ClustersResource.GetRequest getRequest =
+                        Service.Projects.Zones.Clusters.Get(project, zone, cluster);
+                    result = getRequest.Execute();
+                }
+                catch (GoogleApiException apiEx) when (apiEx.HttpStatusCode == HttpStatusCode.NotFound)
+                {
+                    WriteResourceMissingError(
+                        exceptionMessage: $"Cluster '{cluster}' cannot be found in zone '{zone}' of project '{Project}'",
+                        errorId: "ClusterNotFound",
+                        targetObject: cluster);
+                }
+
+                if (result != null)
+                {
+                    yield return result;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns all clusters in zone 'zone' in project 'project'.
+        /// </summary>
+        private IEnumerable<Cluster> GetClustersByZone(string project, string zone)
+        {
+            // This list request does not have page token so we can only send one.
+            ProjectsResource.ZonesResource.ClustersResource.ListRequest listRequest =
+                Service.Projects.Zones.Clusters.List(project, zone);
+            ListClustersResponse response = listRequest.Execute();
+            if (response.Clusters != null)
+            {
+                foreach (Cluster cluster in response.Clusters)
+                {
+                    yield return cluster;
+                }
+            }
+            // The list of clusters returned may be missing these zones.
+            if (response.MissingZones != null)
+            {
+                foreach (string missingZone in response.MissingZones)
+                {
+                    WriteWarning(missingZone);
```

Comment:
We need to add some human readable context to this list of missing clusters. Writing a warning of "us-central1-f" is not very informative.